### DISCUSSION TITLE
[adonisjs 5.0] create ace command examples?

### DISCUSSION THREAD

**Discussion by bstsnail** [adonisjs/core#1539] [Help] [answered]
Hi, I'm new to adonisjs. I want to create a new ace command, but from the help doc, I can only see the doc of adonisjs 4.0. The signature of ace command has changed and I don't know how to use the args and flags.

**Comment by thetutlage** [comment, marked as answer]
Yup. The docs are missing on same. So lemme try to explain it here.

Following command creates a new command for you

```sh
node ace make:command <NAME>
```

The empty command class has 3 properties

```ts
import { BaseCommand } from '@adonisjs/ace'

export default class YourCommandName extends BaseCommand {
	public static commandName = 'commandName'
	public static description = 'Description'

	public async handle() {
	}
}
```

The `commandName` and `description` is for the help screen. The `handle` method is called when your command is executed.

You can accept flags using the `@flags` decorator. Here's an example: https://github.com/adonisjs/core/blob/develop/commands/ListRoutes.ts#L22-L23

You can accept arguments using the `@arg` decorator. Here's an example for this too: https://github.com/adonisjs/assembler/blob/develop/commands/Make/Controller.ts#L37-L38

**Reply by bstsnail** [reply]
thanks， @thetutlage 
I also found the example: https://github.com/adonisjs/ace/blob/develop/example/index.ts
